crates/iroha_executor/src/default/isi/multisig/account.rs

```diff
@@ -0,0 +1,106 @@
+//! Validation and execution logic of instructions for multisig accounts
+
+use iroha_executor_data_model::permission::account::CanRegisterAccount;
+
+use super::*;
+use crate::permission::domain::is_domain_owner;
+
+impl VisitExecute for MultisigRegister {
+    fn visit<V: Execute + Visit + ?Sized>(&self, executor: &mut V) {
+        let registrant = executor.context().authority.clone();
+        let target_domain = self.account.domain();
+        let host = executor.host();
+
+        let Ok(is_domain_owner) = is_domain_owner(target_domain, &registrant, host) else {
+            deny!(
+                executor,
+                "domain must exist before registering multisig account"
+            );
+        };
+
+        let has_permission = {
+            CanRegisterAccount {
+                domain: target_domain.clone(),
+            }
+            .is_owned_by(&registrant, host)
+        };
+
+        // Impose the same restriction as for personal account registrations
+        // TODO Allow the signatories to register the multisig account? With propose and approve procedures?
+        if !(is_domain_owner || has_permission) {
+            deny!(
+                executor,
+                "registrant must have sufficient permission to register an account"
+            );
+        }
+
+        for signatory in self.signatories.keys().cloned() {
+            if host
+                .query(FindAccounts)
+                .filter_with(|account| account.id.eq(signatory))
+                .execute_single()
+                .is_err()
+            {
+                deny!(
+                    executor,
+                    "signatories must exist before registering multisig account"
+                );
+            }
+        }
+    }
+
+    fn execute<V: Execute + Visit + ?Sized>(self, executor: &mut V) -> Result<(), ValidationFail> {
+        let host = executor.host().clone();
```

Comment:
why should you clone the host?

Reply:
Clones implied there was a mutable borrow of the executor while the host was referenced, but I removed them